app/views/errors/technicalDifficulties.scala.html

```diff
@@ -18,22 +18,22 @@
 @import java.time.LocalDateTime
 @import config.ApplicationConfig
 
-@(reference: Option[String], dateTime: LocalDateTime)(implicit requestHeader: RequestHeader, messages: Messages, config: ApplicationConfig)
+@this(mainLayout: views.html.mainLayout)
 
-@main_template(Messages("global.error.InternalServerError500.title"), topNavigation = false) {
+@(reference: Option[String], dateTime: LocalDateTime)(implicit request: Request[_], messages: Messages, config: ApplicationConfig)
 
-    <h1 class="heading-xlarge">@messages("global.error.InternalServerError500.heading")</h1>
+@mainLayout(pageTitle = messages("global.error.InternalServerError500.title"), showTopNavigation = false) {
 
-    <p>@messages("internalServerError.p.1")</p>
+    <h1 class="govuk-heading-xl">@messages("global.error.InternalServerError500.heading")</h1>
 
-    <p>@messages("internalServerError.p.2")</p>
+    <p class="govuk-body">@messages("internalServerError.p.1")</p>
+    <p class="govuk-body">@messages("internalServerError.p.2")</p>
 
-    @reference.map { r =>
-        <p>@messages("internalServerError.reference", r)</p>
+    @for(ref <- reference){
+        <p class="govuk-body">@messages("internalServerError.reference", ref)</p>
```

Comment:
Out of interest, why prefer `for` over `map`?

Reply:
it's a nice pattern for dealing with Option type in Twirl templates. Looks cleaner. A `map` actually maps the Option[A] to an Option[B] and the result is not needed. Instead of `map` a `foreach` should be used as it's only doing a side-effect of writing some HTML to the output. I believe the `for` comprehension will eventually end up a `foreach` because there's no `yield` keyword. It's just nicer like this, and I read somewhere that this is the recommended way to optionally print things in a twirl template.